apps/tasks.py

```diff
@@ -649,3 +503,158 @@ def delete_old_objects():
+        raise self.retry(exc=exc, countdown=10)
+
+
+@worker_ready.connect
+def on_worker_ready(**kwargs):
+    """
+    This function starts the event listener in a given namespace.
+    When the Celery worker is ready, the task is started.
+    """
+    label_selector = "type=app"
+    NAMESPACE = settings.NAMESPACE
+    sync_all_statuses(namespace=NAMESPACE, label_selector=label_selector)
+    init_event_listener.apply_async(args=(NAMESPACE, label_selector), countdown=1)
+
+
+def setup_client():
+    """
+    Sets up the kubernetes python client and event streamer
+    """
+    try:
+        config.load_incluster_config()
+    except config.ConfigException:
+        try:
+            config.load_kube_config(settings.KUBECONFIG)
+        except config.ConfigException:
+            raise config.ConfigException(
+                "Could not set the cluster config. Try to use the cluster.conf file or set incluster config"
+            )
+
+    api = client.CoreV1Api()
+    w = watch.Watch()
+    return api, w
+
+
+def get_status(pod):
+    """
+    Returns the status of a pod by looping through each container
+    and checking the status.
+    """
+    container_statuses = pod.status.container_statuses
+
+    if container_statuses is not None:
+        for container_status in container_statuses:
+            state = container_status.state
+
+            if state is not None:
+                terminated = state.terminated
+
+                if terminated is not None:
+                    reason = terminated.reason
+                    return mapped_status(reason)
+
+                waiting = state.waiting
+
+                if waiting is not None:
+                    reason = waiting.reason
+                    return mapped_status(reason)
+
+                running = state.running
+
+                if running is not None:
+                    return "Running"
+
+            print("Last state not found.")
+
+    return pod.status.phase
+
+
+def mapped_status(reason):
+    status_mapping = {
+        "CrashLoopBackOff": "Error",
+        "Completed": "Retrying...",
+        "ContainerCreating": "Creating",
+        "PodInitializing": "Pending",
+    }
+
+    if reason in status_mapping:
+        return status_mapping[reason]
+    else:
+        return reason
+
+
+def sync_all_statuses(namespace, label_selector):
+    """
+    Syncs the status of all apps with a pod that is on the cluster
+    """
+    api, _ = setup_client()
+    for pod in api.list_namespaced_pod(namespace=namespace, label_selector=label_selector).items:
+        status = pod.status.phase
+        release = pod.metadata.labels["release"]
+        appinstance = AppInstance.objects.filter(parameters__contains={"release": release}).last()
+        if appinstance:
+            status_object = AppStatus(appinstance=appinstance)
+            update_status(appinstance, status_object, status)
+
+
+def update_status(appinstance, status_object, status):
```

Comment:
Considering that there are two updates happening at the same time, shouldn't it be a transaction. Just asking.

Reply:
I wasn't sure, that its needed though..